### PR TITLE
Add Mattermost Role

### PR DESCRIPTION
**Pull request by Eagle-251**
This role will deploy a single Mattermost instance to the Nomad cluster and run a database alongside it.

It is configured to use one CSI volume for both Mattermost and Postgres.
Using Gitlab for authentication can also be configured if desired.
As such, some changes to the Gitlab role are also included as part of this PR.
Those changes have been designed to allow other services to hook into Gitlab for authentication also if the need arises.
Currently, Gitlab SSO requires the use of the Production Lets Encrypt server. A future pull request will seek to fix this issue.

(this PR replaces the previous closed PR that had commits from the wrong branch)

### CODE REVIEW

File: playbooks/roles/mattermost/README.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    minsize: 10737418240 
+    maxsize: 32212254720
+    namespace: default
+    type: csi 
+    filesystem: ext4
+    accessmode: single-node-writer
+    attachmentmode: file-system
+    read_only: "false"
+```
+
```
Comment (sejo) [inline playbooks/roles/mattermost/README.md:57]: What if you don't use CSI, but local install or on local hardware with nfs backend?

File: playbooks/roles/mattermost/templates/mattermost.hcl.j2 [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        {% if 'aarch64' != ansible_architecture %}
+        logging {
+          type = "loki"
+        }
+        {% endif %}
+      }
+
+    }
+
+    task "mattermost_db" {
```
Comment (sejo) [inline playbooks/roles/mattermost/templates/mattermost.hcl.j2:109]: No logging here?

**Review by sejo — CHANGES_REQUESTED** [review]

File: playbooks/roles/mattermost/README.md (hunk 1 above)
Comment (Eagle-251) [reply playbooks/roles/mattermost/README.md:57]: I am going to fix this in an upcoming PR to update Molecule compatibility by initially using Nomad host volumes. I could also investigate including NFS.

Comment (sejo) [reply playbooks/roles/mattermost/README.md:57]: ok, did you create a ticket for that yet?

Comment (Eagle-251) [reply playbooks/roles/mattermost/README.md:57]: Just created one now - #26

**Pull request merged**